src/audio/component.c

```diff
@@ -95,6 +95,33 @@ void comp_unregister(struct comp_driver *drv)
 	spin_unlock(&cd->lock);
 }
 
+int comp_set_sink_buffer(struct comp_dev *dev, uint32_t period_bytes,
+			 uint32_t periods)
+{
+	struct comp_buffer *sinkb;
+	int ret = 0;
+
+	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer,
+				source_list);
+
+	/* components connected to dma configure buffer independently */
+	if (!sinkb->sink->is_dma_connected) {
+		ret = buffer_set_size(sinkb, period_bytes * periods);
+		if (ret < 0) {
+			trace_comp_error("comp_set_sink_buffer() error: "
+					 "buffer_set_size() failed");
+			return ret;
+		}
+	} else if (sinkb->size < period_bytes * periods) {
```

Comment:
This check is invalid. `buffer_set_size()` can reallocate buffer to bigger size.

Reply:
My intention was not to increase buffer size, when sink component is connected to dma, because in that case sink component reconfiguration would be required.